cmd/auth/main.go

```diff
@@ -37,7 +38,11 @@ func main() {
 		AuthInfoStore: &auth.AuthInfoStoreProvider{CanMigrate: true},
 	}
 
-	srv := server.NewServer("localhost:3000", configuration.DevMode)
+	srv := server.NewServer("localhost:3000")
+
+	if configuration.DevMode {
+		srv.Use(middleware.EnvTenantMiddleware{}.Handle)
+	}
```

Comment:
My preference is still use Hanlder(https://golang.org/pkg/net/http/#Handler) as middleware, not mux.MiddlewareFunc.

Reply:
Do you mean like `StripPrefix` in https://golang.org/pkg/net/http/#StripPrefix?

But `net/http` itself is not consistent, see `TimeoutHandler` and `StripPrefix`.

Reply:
Or do you mean like `CORSMiddleware` in skygear-server?

Reply:
I mean `CORSMiddleware` in skygear-server.

Reply:
I prefer `mux.MiddlewareFunc`. 

Middleware and handler are practically not exchangeable, because take skygear-server middlewares as example, they all require the `Next` to be not nil. So if someone create a `CORSMiddleware` but does not pass a `Next` handler to it, it may create runtime error. Making it declarative (i.e. use something like a `mux.MiddlewareFunc` interface) can prevent this, also the code look cleaner.

node express and elixir phoenix also take a declarative approach on the middleware.